### PR TITLE
fix: web terminal outside argocd namespace (#11166)

### LINKED ISSUE #11166: Exec in apps outside of ArgoCD namespace not working
Together with the 2.5.0 update I deployed all ArgoCD apps apart from ArgoCD itself to their respective namespaces. When I want to exec into an ArgoCD pod everything works as usual. However when I try to exec into a pod in a different namespace I receive the error `Terminal Connection Error: Internal error`.

My policy.csv

```
p, admin, exec, create, */*, allow
g, group-name, admin
```

Checking the RBAC permissions for the argocd-server serviceaccount everything also looks good.

```
kubectl auth can-i create pods --subresource='exec' --as=system:serviceaccount:argocd:argocd-server
yes
```

The logs unfortunately don't print anything useful.

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Fixes #11166

### CODE REVIEW

**Review by jannfis — COMMENTED** [review]
Some comments for discussion, please see below.

File: pkg/apis/application/v1alpha1/types.go [hunk 1]
```diff
@@ -2575,9 +2575,13 @@ func (a *Application) QualifiedName() string {
 // RBACName returns the full qualified RBAC resource name for the application
 // in a backwards-compatible way.
 func (a *Application) RBACName(defaultNS string) string {
-	if defaultNS != "" && a.Namespace != defaultNS && a.Namespace != "" {
-		return fmt.Sprintf("%s/%s/%s", a.Spec.GetProject(), a.Namespace, a.Name)
+	return AppRBACName(defaultNS, a.Spec.GetProject(), a.Namespace, a.Name)
+}
+
+func AppRBACName(defaultNS string, project string, namespace string, name string) string {
```
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply pkg/apis/application/v1alpha1/types.go:2581]: Would putting it in a utils module make more sense?

File: util/security/application_namespaces.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
+package security
+
+import (
+	"fmt"
+
+	"github.com/argoproj/argo-cd/v2/util/glob"
+)
+
+func IsNamespaceEnabled(namespace string, serverNamespace string, enabledNamespaces []string) bool {
```
Comment (jannfis) [inline util/security/application_namespaces.go:9]: Do we want to add some tests? :)

File: util/security/rbac.go [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package security
+
+import (
+	"fmt"
+)
+
+// AppRBACName constructs name of the app for use in RBAC checks.
+func AppRBACName(defaultNS string, project string, namespace string, name string) string {
```
Comment (jannfis) [inline util/security/rbac.go:8]: Do we want to add some tests?

**Review by jannfis — APPROVED** [review]
LGTM! Thanks!

**Comment by quentinlacrouts** [comment]
Hi, any chance that can be merged this week ? 
Thanks

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.6 for 2.6.0-rc2 and release-2.5 for 2.5.6.

**Pull request merged**